docs/core/diagnostics/trace-perfcollect-lttng.md

```diff
@@ -0,0 +1,247 @@
+
+1. Download `perfcollect`.
+
+    > ```bash
+    > curl -OL http://aka.ms/perfcollect
+    > ```
+
+2. Make the script executable.
+
+    > ```bash
+    > chmod +x perfcollect
+    > ```
+
+3. Install tracing prerequisites - these are the actual tracing libraries.
+
+    > ```bash
+    > sudo ./perfcollect install
+    > ```
+
+    This will install the following prerequisites on your machine:
+
+    1. `perf`: the Linux Performance Events sub-system and companion user-mode collection/viewer application. `perf` is part of the Linux kernel source, but is not usually installed by default.
+
+    2. `LTTng`: Used to capture event data emitted at runtime by CoreCLR. This data is then used to analyze the behavior of various runtime components such as the GC, JIT and thread pool.
+
+## Collecting a Trace
+
+1. Have two shells available - one for controlling tracing, referred to as **[Trace]**, and one for running the application, referred to as **[App]**.
+
+2. **[Trace]** Start collection.
+
+    > ```bash
+    > sudo ./perfcollect collect sampleTrace
+    > ```
+
+    Expected Output:
+
+    > ```bash
+    > Collection started.  Press CTRL+C to stop.
+    > ```
+
+3. **[App]** Set up the application shell with the following environment variables - this enables tracing configuration of CoreCLR.
+
+    > ```bash
+    > export COMPlus_PerfMapEnabled=1
+    > export COMPlus_EnableEventLog=1
+    > ```
+
+4. **[App]** Run the app - let it run as long as you need to in order to capture the performance problem. The exact length can be as short as you need as long as it sufficiently captures the window of time where the performance problem you want to investigate occurs.
+
+    > ```bash
+    > dotnet run
+    > ```
+
+5. **[Trace]** Stop collection - hit CTRL+C.
+
+    > ```bash
+    > ^C
+    > ...STOPPED.
+    >
+    > Starting post-processing. This may take some time.
+    >
+    > Generating native image symbol files
+    > ...SKIPPED
+    > Saving native symbols
+    > ...FINISHED
+    > Exporting perf.data file
+    > ...FINISHED
+    > Compressing trace files
+    > ...FINISHED
+    > Cleaning up artifacts
+    > ...FINISHED
+    >
+    > Trace saved to sampleTrace.trace.zip
+    > ```
+
+    The compressed trace file is now stored in the current working directory.
+
+## Viewing a Trace
+
+There are number of options for viewing the trace that was collected. Traces are best viewed using [PerfView](http://aka.ms/perfview>) on Windows, but they can be viewed directly on Linux using `PerfCollect` itself or `TraceCompass`.
+
+Recent versions of .NET Core and the Linux perf tool support automatic resolution of method names for framework code. If you are working with .NET Core version 3.1 or less, an extra step is necessary. See [Resolving Framework Symbols](#resolving-framework-symbols) for details.
```

Comment:
I'd suggest moving these callouts about resolving framework and runtime symbols upwards to the 'preparing your machine' section. If the user reads this in order then they have already collected the trace by this point. It is too late for them to act on these new instructions without starting over.